assets/js/admin/course-theme-sidebar.js

```diff
@@ -3,49 +3,147 @@
  */
 import { PluginDocumentSettingPanel } from '@wordpress/edit-post';
 import { __ } from '@wordpress/i18n';
-import { SelectControl } from '@wordpress/components';
+import { ToggleControl } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
 import useCourseMeta from '../../react-hooks/use-course-meta';
+import { name as courseOutlineBlockName } from '../../blocks/course-outline/outline-block/block.json';
+import { name as courseModuleBlockName } from '../../blocks/course-outline/module-block/block.json';
+import { name as courseLessonBlockName } from '../../blocks/course-outline/lesson-block/block.json';
 
 const SENSEI_THEME = 'sensei-theme';
 const WORDPRESS_THEME = 'wordpress-theme';
+const SENSEI_PREVIEW_QUERY = 'sensei_theme_preview';
+
+const canPreview = ( block ) =>
+	block.name === courseLessonBlockName &&
+	block.attributes.id &&
+	! block.attributes.draft;
 
 /**
  * Course Theme Sidebar component.
  */
 const CourseThemeSidebar = () => {
 	const [ theme, setTheme ] = useCourseMeta( '_course_theme' );
 
+	const globalLearningModeEnabled =
+		window.senseiSettings.sensei_learning_mode_all || false;
+	const currentPost = useSelect( ( select ) =>
```

Comment:
Maybe extracting all this part that gets the `customizerUrl` and `previewUrl` would make this component cleaner. It could be a React Hook inside this same file since it's used only here. Does it make sense?

Reply:
Done. dbf9b20ea640685ec546bcf62b0ddfe52c5c4fc1